### PR TITLE
#73. Fix bugs from Firebase Crashlytics.

### PR DESCRIPTION
**Pull request by bogdan-evtushenko**

### CODE REVIEW

**Comment by bogdan-evtushenko** [comment]
Oh, okey)

File: app/src/main/java/com/bogdan/codeforceswatcher/features/users/UsersAdapter.kt [hunk 1]
```diff
@@ -87,7 +87,7 @@ class UsersAdapter(
 
         init {
             view.setOnClickListener {
-                itemClickListener.invoke(adapterPosition)
+                if (adapterPosition != RecyclerView.NO_POSITION) itemClickListener.invoke(adapterPosition)
```
Comment (bogdan-evtushenko) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/users/UsersAdapter.kt]: Check https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.ViewHolder.html#getAdapterPosition()
One user has a crash connected to it.

File: app/src/main/java/com/bogdan/codeforceswatcher/features/users/redux/reducers/UsersReducer.kt [hunk 2]
```diff
@@ -15,12 +15,13 @@ fun usersReducer(action: Action, state: AppState): UsersState {
...
             )
         }
+
         is UsersRequests.FetchUsers.Success -> {
-            newState = newState.copy(
-                status = UsersState.Status.IDLE,
-                users = action.users
-            )
+            val mapUsers = action.users.associateBy { it.id }
+            val newUsers = state.users.users.map { mapUsers[it.id] ?: it }
```
Comment (bogdan-evtushenko) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/users/redux/reducers/UsersReducer.kt]: Try to swipe users list for update and in time(while users are fetching) delete one user - he appears back after fetching(but in db he is already removed). Of course if you try to remove user second time - application will crash.

**Comment by bogdan-evtushenko** [comment]
Done.

File: app/src/main/java/com/bogdan/codeforceswatcher/features/users/redux/reducers/UsersReducer.kt (hunk 2 above)
Comment (yevhenii-kanivets) [reply app/src/main/java/com/bogdan/codeforceswatcher/features/users/redux/reducers/UsersReducer.kt]: Tricky bug 👀I think, your fix should work 👍

**Pull request merged**